spec/lib/tasks/location_children_count_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rake'
+
+RSpec.describe 'location:fix_children_count' do
+  before do
+    Rails.application.load_tasks
+    @parent = create(:location)
+    @child1 = create(:location, parent: @parent)
+    @child2 = create(:location, parent: @parent)
+    @grandchild = create(:location, parent: @child1)
+
+    # Manually set children_count to 0 for all locations
+    # rubocop:disable Rails/SkipsModelValidations
+    @parent.update_column(:children_count, 0)
+    @child1.update_column(:children_count, 0)
+    @child2.update_column(:children_count, 0)
+    # rubocop:enable Rails/SkipsModelValidations
+  end
+
+  it 'fixes the children_count for all locations and outputs the correct message' do
+    expect do
+      Rake::Task['location:fix_children_count'].invoke
```

Comment:
I would add expectations to confirm that children_count is 0 before running the rake tests as otherwise we can't know for sure that children_count is actually updated.